services/data/src/hooks/useDataQuery.ts

```diff
@@ -33,20 +33,24 @@ const fetchData = (
     )
 }
 
-export const useDataQuery = (query: QueryMap): QueryRenderInput => {
+export const useDataQuery = (initialQuery: QueryMap): QueryRenderInput => {
     const context = useContext(DataContext)
+    const query = useRef<QueryMap>(initialQuery)
     const [state, setState] = useState<QueryState>({ loading: true })
     const [refetchCount, setRefetchCount] = useState(0)
     const refetch: RefetchCallback = useCallback(
-        () => setRefetchCount(count => count + 1),
+        (newQuery: QueryMap | undefined) => {
+            if (newQuery) query.current = newQuery
```

Comment:
OPTIONAL: If we wanted to clean this up even more, we could create a *deep clone* (possibly also performing a *deep freeze*) of the query object here as well as on initiation so that we *always* update the value of `query` when `refetch` is called.  This would actually mean we could get rid of `refetchCount` altogether, and has the added benefit of protecting the `query` object from external mutation, but it would come at a slight initial performance hit (though probably negligible)